src/metabase/api/recent_errors.clj

```diff
@@ -0,0 +1,25 @@
+(ns metabase.api.recent-errors
+  (:require [amalloy.ring-buffer :as ring-buffer]
+            [compojure.core :refer [GET]]
+            [metabase.api.common :as api]))
+
+(set! *warn-on-reflection* true)
+
+(def ^:private recent-errors (atom (ring-buffer/ring-buffer 10)))
+
+(defn record!
+  "Records an exception and the time it occurred so we can tell the frontend about it later."
+  [e]
+  (swap! recent-errors conj {:error e
+                             :occurred-at (java.time.Instant/now)}))
+
+;; record an exception for dev
+(record! (ex-info "meow mix meow mix we deliver" {}))
```

Comment:
Should this be removed?

Reply:
no way, this is the best part

Reply:
cc @johnswanson

Reply:
Oops, we should remove this entire namespace.

Reply:
[This commit](https://github.com/metabase/metabase/pull/39817/commits/25d53dc9dfc2021d35efca190bd4aeafbaa6135b) is the only one we actually want.